auth/providers/azure/graph/graph.go

```diff
@@ -30,12 +30,19 @@ import (
 	jsoniter "github.com/json-iterator/go"
 	"github.com/moul/http2curl"
 	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 )
 
 // These are the base URL endpoints for MS graph
 var (
 	json                  = jsoniter.ConfigCompatibleWithStandardLibrary
 	expandedGroupsPerCall = 500
+
+	getMemberGroupsFailed = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "guard_graph_failure_total",
```

Comment:
should guard_graph_failure_total have guard_ prefix?

nonetheless, recommend guard_azure_graph_failure_total for azure specific counter